AI/MMAI/BAI/model/NNModel.cpp

```diff
@@ -0,0 +1,868 @@
+/*
+ * NNModel.cpp, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+
+#include <array>
```

Comment:
Any reasons for adding these includes here?
Most (or maybe all) of them should come from `Global.h` (-> included by `StdInc.h` which you already added)